[Xavierlhm] iP

Xavierlhm opened the pull request:
# Tracker

Tracker is a program that helps keep track of your tasks, so that you can focus on other more important things 😉.

## Features

- [x] Add new tasks as todo, deadline, or event.
- [x] Delete tasks.
- [x] Mark tasks as done.
- [x] Mark tasks as undone.
- [x] Filter tasks with keyword.

## Installation

1. Please refer to the installation instructions on the [download](https://github.com/Xavierlhm/ip/releases) page.

## Testing

1. Once you download the .jar file, execute the following command to run the application:
```ruby 
java -jar tracker.jar
```

## Sneak Peek
> Below is the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

## Acknowledgements
**(These are the people who did the hard stuff)**
- ~~Myself~~ :cry:

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Xavierlhm committed:
Renamed Duke.java to Tracker.java and added codes to rename, greet, exit

Xavierlhm committed:
Added codes to echo user input

Xavierlhm committed:
Added codes to add and list user input

Xavierlhm committed:
Added Task.java with codes to mark task as done or undone

Xavierlhm committed:
Added Deadline.java, Event.java, Todo.java to support tracking for three types of tasks

Xavierlhm committed:
Modified EXPECTED.TXT, input.txt, runtest.bat to use input/output redirection technique to semi-automate the testing of Tracker

Xavierlhm committed:
Added TrackerException.java to handle errors

Xavierlhm committed:
Modified Tracker.java to support deleting of tasks from the list
Modified Todo.java to display output correctly

Xavierlhm committed:
Modified Deadline.java, Event.java, Todo.java, Task.java to use Java Enums
Added TaskType.java to aid in usage of Java Enums

Xavierlhm committed:
Add TaskManager.java

Xavierlhm committed:
Modify TaskManager.java

Xavierlhm committed:
Merge branch 'branch-Level-7'

Xavierlhm committed:
Modify Tracker.java, TaskManager.java, Deadline.java, Event.java

Xavierlhm committed:
Merge branch 'branch-Level-8'

Xavierlhm committed:
Add Ui.java, Storage.java, Parser.java, TaskList.java

Xavierlhm committed:
Merge branch 'branch-A-MoreOOP'

Xavierlhm committed:
Move all classes into one package tracker

Xavierlhm committed:
Merge branch 'branch-A-Packages'

Xavierlhm committed:
Merge branch 'add-gradle-support'

Xavierlhm committed:
Add add-gradle-support

Xavierlhm committed:
Merge branch 'master' of https://github.com/Xavierlhm/ip

# Conflicts:
#	gradlew
#	gradlew.bat

Xavierlhm committed:
Add TaskListTest.java and AddTodoCommandTest.java

Xavierlhm committed:
Merge branch 'branch-A-JUnit'

Xavierlhm committed:
Package the App as a JAR file

Xavierlhm committed:
Merge branch 'branch-A-Jar'

Xavierlhm committed:
Add JavaDoc to classes and methods

Xavierlhm committed:
Merge branch 'branch-A-JavaDoc'

Xavierlhm committed:
Tweak the code to comply with coding standard

Xavierlhm committed:
Merge branch 'branch-A-CodingStandard'

Xavierlhm committed:
Add FindCommand.java

Xavierlhm committed:
Modify codes to follow checkstyle

Xavierlhm committed:
Merge branch 'branch-A-CheckStyle'

Xavierlhm committed:
Add Launcher, Main, MainWindow, DialogBox

izniy commented on src/main/java/tracker/FindCommand.java:
```diff
@@ -0,0 +1,45 @@
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
+        StringBuilder response = new StringBuilder("Here are the matching tasks in your list:");
+        boolean found = false;
```
Can be changed to sound like a boolean, like isFound

izniy commented on src/main/java/tracker/Task.java:
```diff
@@ -0,0 +1,104 @@
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
Good naming of the boolean!

izniy reviewed (commented):
LGTM. Just some nits to fix. Good job :)

Xavierlhm committed:
Add assert statements

Xavierlhm committed:
Merge branch 'branch-A-Assertions'

Xavierlhm committed:
Improve code quality

Xavierlhm committed:
Merge branch 'branch-A-CodeQuality'

Xavierlhm committed:
Follow coding standard and quality

There are several methods and codes that uses magic numbers to perform its logic.

As a step towards better code quality, let's remove the use of magic numbers and avoid complicated expressions.

Xavierlhm committed:
Add new extension

Xavierlhm committed:
Use Java Streams in FindCommand.java

Xavierlhm committed:
Merge pull request #1 from Xavierlhm/branch-A-Streams

Use Java Streams in FindCommand.java

Xavierlhm committed:
Use Java Streams and tidy up code

geeliette commented on src/main/java/tracker/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,62 @@
+        assert ui != null : "UI cannot be null";
+        StringBuilder response = new StringBuilder();
+        String[] parts = input.substring(SPLIT_INDEX).split(" /by ");
+        boolean isLessThanLimit = parts.length < MAX_SIZE;
+        boolean isDescriptionEmpty = parts[FIRST_PART].trim().isEmpty();
+        boolean isMoreThanLimit = parts.length >= MAX_SIZE;
+        boolean isDateEmpty = isMoreThanLimit && parts[SECOND_PART].trim().isEmpty();
+        boolean isValidCode = isLessThanLimit || isDescriptionEmpty || isDateEmpty;
```
I like how you named all your booleans appropriately

geeliette commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,10 @@
+todo read book
```
Nice job on testing for your cases! Maybe you could try testing for errors as well, eg. test to see if your bot responds correctly when user writes an invalid input

geeliette commented on src/main/java/tracker/DialogBox.java:
```diff
@@ -0,0 +1,89 @@
+        displayPicture.setImage(img);
+    }
+
+    private void changeDialogStyle(String commandType) {
+        switch (commandType) {
+        case "AddCommand":
+            dialog.getStyleClass().add("add-label");
+            break;
+        case "ChangeMarkCommand":
+            dialog.getStyleClass().add("marked-label");
+            break;
+        case "DeleteCommand":
+            dialog.getStyleClass().add("delete-label");
+            break;
+        default:
+            // Do nothing
```
Nice one on following the indentations for case clauses for the course's coding standard

geeliette reviewed (commented):
LGTM, just a few nits to fix

Xavierlhm committed:
Add transparent background

Xavierlhm committed:
Merge branch 'branch-A-BetterGui'

Xavierlhm committed:
Add new phrase

Xavierlhm committed:
Merge branch 'branch-A-Personality'

Xavierlhm committed:
Add function to check if file exist

Xavierlhm committed:
Merge branch 'branch-A-MoreErrorHandling'

Xavierlhm committed:
Add User Guide

Xavierlhm committed:
Merge branch 'branch-A-UserGuide'

Xavierlhm committed:
Create a new JAR file

Xavierlhm committed:
Merge branch 'branch-A-Release'

Xavierlhm committed:
Modify code to adhere to SLAP principle

Xavierlhm committed:
Merge branch 'branch-SLAP'

Xavierlhm committed:
Add javadoc to new methods

Xavierlhm committed:
Merge branch 'branch-TouchUp'

Xavierlhm committed:
Modify AddTodoCommandTest

Xavierlhm committed:
Modify user guide

Xavierlhm committed:
Merge branch 'branch-Update-UG'

Xavierlhm committed:
Modify user guide

Xavierlhm committed:
Modify user guide
